### DISCUSSION TITLE
WSL2: Tauri is not creating my executable

### DISCUSSION THREAD

**Discussion by do-kevin** [tauri-apps/tauri#6207] [Q&A] [answered]
Hi folks,

I'm completely new to this and just learning how to create executables with JavaScript. Sorry if this is a more of a WSL issue. I am running on Windows 10. So I just used `create-tauri-app` and set `targets` to `"msi"`. The compilation successfully completes, but there's no console.log or physical production of the executable I wanted to create and test. I'm following [elibro's Tauri & ReactJS - Creating Modern Desktop Apps](https://www.youtube.com/watch?v=BbZmLXBDGnU&list=PLmWYh0f8jKSjt9VC5sq2T3mFETasG2p2L) video, but it's a year old. His video shows the .exe being created.

Is this the proper behavior? Or is there something work with my workspace and I am missing some packages to install?

**Comment by FabianLars** [comment, marked as answer]
WSL is basically Linux and since Tauri does not support cross-compilation yet, WSL can only produce Linux app bundles (.deb, .appimage).

elibro used windows' plain old powershell in their video, not WSL.

**Reply by do-kevin** [reply]
Thank you, I'll just try out Powershell. Any problems if I run Power Shell with Git Bash to compile the .exe or doesn't matter or isn't related?

**Reply by FabianLars** [reply]
Shouldn't matter as git bash is still Windows, but there were issues with git bash in the past not being able to compile at all or something 🤔 idk just try it out and if it doesn't work go back to plain powershell/cmd.exe

**Reply by do-kevin** [reply]
> Shouldn't matter as git bash is still Windows, but there were issues with git bash in the past not being able to compile at all or something 🤔 idk just try it out and if it doesn't work go back to plain powershell/cmd.exe

So I followed the prerequisites and installed what was needed (https://tauri.app/v1/guides/getting-started/prerequisites). And I set the targets to just `msi`. On Powershell, Developer Powershell for 2022, and Git Bash, it keeps building just the html, JS, and CSS when I run `yarn build`. And the `target` folder does not appear under my `tsc-tauri`.

Here's my repo: https://github.com/do-kevin/tauri-practice. It's basically just the create tauri-app.

Is there some additional setup I am missing?

**Reply by FabianLars** [reply]
`yarn build` indeed only builds your frontend (just like `yarn dev` will only start your frontend in dev mode), you're looking for `yarn tauri build`

**Reply by do-kevin** [reply]
🤦Ah my goodness, one week past by and I forgot all about `tauri build`. Thank you so much; it's compiling for me now since I got the pre-reqs down and I ran the command in Powershow (admin).

**Comment by elibroftw** [comment]
It's funny googling something and someone talking about me in their question. 
I needed to build something in WSL as well so here is how I did it.

1. Ensure you have WSL 2 installed. Use `wsl --list -v` to check. If you don't, I suggest first setting default wsl version to 2 and then uninstalling the distro and reinstalling it. Backup important files
2. After installing tauri pre-requisitues, follow https://penner.me/getting-tauri-working-in-wsl (installing graphics libraries)

```sh
sudo apt install pkg-config \
  libdbus-1-dev \
  libgtk-3-dev \
  libsoup2.4-dev \
  libjavascriptcoregtk-4.0-dev \
  libwebkit2gtk-4.0-dev -y
echo "export DISPLAY=:0" >> ~/.bashrc
```

You may also need vcxsrv installed as I can see the UI on my PC which has it installed and not my laptop which does not have it installed.
